libhoney/transmission.py

```diff
@@ -84,19 +84,33 @@ def _send(self, ev):
             "X-Honeycomb-Team": ev.writekey,
             "X-Honeycomb-SampleRate": str(ev.sample_rate)})
         preq = self.session.prepare_request(req)
-        resp = self.session.send(preq)
-        if (resp.status_code == 200):
-            sd.incr("messages_sent")
-        else:
+        try:
+            resp = self.session.send(preq)
+            if (resp.status_code == 200):
+                sd.incr("messages_sent")
+            else:
+                sd.incr("send_errors")
+            dur = get_now() - start
+            response = {
+                "status_code": resp.status_code,
+                "duration": dur.total_seconds() * 1000,  # report in milliseconds
+                "metadata": ev.metadata,
+                "body": resp.text,
+                "error": "",
+            }
+        except requests.exceptions.ConnectionError as e:
```

Comment:
Why catch only this type of error? I think right now, any other kind of error thrown will kill the thread, diminishing the size of the thread pool. If that happens enough times the thread pool will end up empty, in which case libhoney will silently stop doing anything except consuming memory as the queue fills up.